test/unit/StreamrClient.test.js

```diff
@@ -50,10 +50,10 @@ describe('StreamrClient', () => {
         asyncs = []
     }
 
-    function setupSubscription(streamId, emitSubscribed = true, subscribeOptions = {}, handler = sinon.stub()) {
+    async function setupSubscription(streamId, emitSubscribed = true, subscribeOptions = {}, handler = sinon.stub()) {
         assert(client.isConnected(), 'setupSubscription: Client is not connected!')
         connection.expect(new SubscribeRequest(streamId))
-        const sub = client.subscribe({
+        const sub = await client.subscribe({
```

Comment:
See comment on `subscribe` - we should keep returning the `Subscription` to avoid breaking things. Just commenting here to detect the change